src/dataflow/sink/kafka.rs

```diff
@@ -45,69 +47,78 @@ pub fn kafka<G>(
         Ok(schema_id) => {
             let mut config = ClientConfig::new();
             config.set("bootstrap.servers", &connector.url.to_string());
-            let admin: AdminClient<DefaultClientContext> =
-                config.create().expect("creating admin kafka client failed");
 
-            let admin_opts = AdminOptions::new().operation_timeout(Some(Duration::from_secs(5)));
-            let new_topic = NewTopic::new(&connector.topic, 1, TopicReplication::Fixed(1));
-            let res = block_on(admin.create_topics(&[new_topic], &admin_opts));
-            match res {
-                Ok(res) => {
-                    if res.len() != 1 {
-                        error!(
+            if write_to_kafka {
```

Comment:
This relies on differential implementation details, right? Specifically that the exchange below will route all data to the same worker for which `write_to_kafka` is true. Otherwise there's a auto-topic-creation race if the worker that creates the Kafka topic is not the worker that produces messages, because the message production worker could start firing messages off first and trigger auto-creation of the topic with the wrong settings, like multiple partitions. (This bit us a lot with testdrive.)

We might want to pass the scope in to this function, so we can call `scope.index()` in here and compute `write_to_kafka` in here, just to keep the code that deals with this delicate dance in the same place and slap a big comment to the effect of `WARNING WARNING relying on internal differential details about how Exchange routes data to workers` on top.

Does that make sense? It's late and I'm tired.

Reply:
That does make sense! I think the requested diff + comment here is pretty light actually and will be a nice convention for file sinks too